cpp/src/arrow/type_traits.h

```diff
@@ -22,7 +22,7 @@
 #include <type_traits>
 #include <vector>
 
-#include "arrow/type_fwd.h"
+#include "arrow/type.h"
```

Comment:
this change seems odd; was type_fwd incorrect?

Reply:
Yes, there are errors such as:
```
../src/arrow/type_traits.h:149:1: error: incomplete type 'arrow::UInt8Type' named in nested name specifier
PRIMITIVE_TYPE_TRAITS_DEF(uint8_t, UInt8, uint8)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/arrow/type_traits.h:145:14: note: expanded from macro 'PRIMITIVE_TYPE_TRAITS_DEF'
      CType, ARROW_CONCAT(ArrowShort, Type), ARROW_CONCAT(ArrowShort, Array), \
      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/arrow/util/macros.h:24:28: note: expanded from macro 'ARROW_CONCAT'
#define ARROW_CONCAT(x, y) x##y
                           ^
<scratch space>:126:1: note: expanded from here
UInt8Type
^
../src/arrow/type_traits.h:130:19: note: expanded from macro 'PRIMITIVE_TYPE_TRAITS_DEF_'
    using CType = ArrowType_::c_type;                                                    \
                  ^~~~~~~~~~~~
../src/arrow/type_fwd.h:186:1: note: forward declaration of 'arrow::UInt8Type'
_NUMERIC_TYPE_DECL(UInt8)
^
../src/arrow/type_fwd.h:176:9: note: expanded from macro '_NUMERIC_TYPE_DECL'
  class KLASS##Type;                                  \
        ^
<scratch space>:274:1: note: expanded from here
UInt8Type
^
```